### PR TITLE
[numpy] fix mixed type backward

### PR DESCRIPTION
**Pull request by BenjaminCHEN2016**
## Description ##

attempt to fix issue regarding mixed type in binary op add, minus, true_divide, mod
#18022 #18084 

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at https://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

File: src/operator/mshadow_op.h [hunk 1]
```diff
@@ -746,6 +746,10 @@ MXNET_BINARY_MATH_OP_NC(minus_sign, a - b > DType(0) ? DType(1) : -DType(1));
 
 MXNET_BINARY_MATH_OP(rminus, b - a);
 
+MXNET_BINARY_MATH_OP_NC(self_grad, 1);
+
+MXNET_BINARY_MATH_OP_NC(minus_rgrad, -1);
+
```
Comment (hgt312) [inline src/operator/mshadow_op.h:752]: Could use better func name? Like `posone`, `negone`?

File: src/operator/numpy/np_elemwise_broadcast_op.cc [hunk 2]
```diff
@@ -115,7 +115,22 @@ MXNET_OPERATOR_REGISTER_NP_BINARY_MIXED_PRECISION(_npi_add)
...
-.set_attr<nnvm::FGradient>("FGradient", ElemwiseGradUseNone{"_backward_broadcast_add"});
+.set_attr<nnvm::FGradient>("FGradient", ElemwiseGradUseIn{"_backward_npi_broadcast_add"});
+
+NNVM_REGISTER_OP(_backward_npi_broadcast_add)
+.set_num_inputs(3)
+.set_num_outputs(2)
+.set_attr<nnvm::TIsBackward>("TIsBackward", true)
+.set_attr<nnvm::FInplaceOption>("FInplaceOption",
+  [](const NodeAttrs& attrs){
+    return std::vector<std::pair<int, int> >{{0, 1}};
```
Comment (hgt312) [inline src/operator/numpy/np_elemwise_broadcast_op.cc:126]:
```suggestion
    return std::vector<std::pair<int, int> >{{0, 0}, {0, 1}};
```

**Comment by BenjaminCHEN2016** [comment]
@mxnet-bot run ci [miscellaneous, windows-cpu, windows-gpu]

**Comment by BenjaminCHEN2016** [comment]
@mxnet-bot run ci [windows-cpu, windows-gpu]

**Comment by BenjaminCHEN2016** [comment]
@mxnet-bot run ci [unix-gpu]

**Comment by hgt312** [comment]
LGTM. @yzhliu

**Comment by yzhliu** [comment]
Thanks @BenjaminCHEN2016 @hgt312

**Pull request merged**